tests/jobs/localisation/example_case/scripts/common_functions.py

```diff
@@ -0,0 +1,526 @@
+        for indices in coarse_cell_index_list:
+            Iindx = indices[0] - 1
+            Jindx = indices[1] - 1
+            Kindx = indices[2] - 1
+            istart = mx * Iindx
+            iend = istart + mx
+            jstart = my * Jindx
+            jend = jstart + my
+            kstart = mz * Kindx
+            kend = kstart + mz
+            sum_val = 0.0
+            for k in range(kstart, kend):
+                for j in range(jstart, jend):
+                    for i in range(istart, iend):
+                        sum_val += field_values[i, j, k]
+            upscaled_values[Iindx, Jindx, Kindx] = sum_val / (mx * my * mz)
+
+    return upscaled_values
+
+
+def trend():
+    """
+    Return 3D numpy array with values following a linear trend
+    scaled to take values between 0 and 1.
+    """
+    nx, ny, nz = settings.field.grid_dimension
+    xsize = settings.grid_size.xsize
+    ysize = settings.grid_size.ysize
+    a, b = settings.field.trend_params
+
+    x0 = 0.0
+    y0 = 0.0
+    dx = xsize / nx
+    dy = ysize / ny
+
+    maxsize = ysize
+    if xsize > ysize:
+        maxsize = xsize
+
+    val = np.zeros((nx, ny, nz), dtype=np.float32, order="F")
+    for i in range(nx):
+        x = x0 + i * dx
+        for j in range(ny):
+            y = y0 + j * dy
+            for k in range(nz):
+                val[i, j, k] = a * (x - x0) / maxsize + b * (y - y0) / maxsize
+
+    minval = np.min(val)
+    maxval = np.max(val)
+    val_normalized = (val - minval) / (maxval - minval)
+    return val_normalized
+
+
+def simulate_field(start_seed):
+    # pylint: disable=no-member,import-outside-toplevel
+    # This function will not be available untill gaussianfft is available on python 3.10
+    # import gaussianfft as sim  # isort: skip
+    # dummy code to avoid pylint complaining in github actions
+    sim = None
+
+    variogram_name = settings.field.variogram
+    corr_ranges = settings.field.correlation_range
+    azimuth = settings.field.correlation_azimuth
+    dip = settings.field.correlation_dip
+    alpha = settings.field.correlation_exponent
+    nx, ny, nz = settings.field.grid_dimension
+    xsize = settings.grid_size.xsize
+    ysize = settings.grid_size.ysize
+    zsize = settings.grid_size.zsize
+
+    xrange = corr_ranges[0]
+    yrange = corr_ranges[1]
+    zrange = corr_ranges[2]
+
+    dx = xsize / nx
+    dy = ysize / ny
+    dz = zsize / nz
+
+    print(f"Start seed: {start_seed}")
+    sim.seed(start_seed)
+
+    variogram = sim.variogram(
+        variogram_name,
+        xrange,
+        perp_range=yrange,
+        depth_range=zrange,
+        azimuth=azimuth - 90,
+        dip=dip,
+        power=alpha,
+    )
+
+    print(f"Simulate field with size: nx={nx},ny={ny} ")
+    field1D = sim.simulate(variogram, nx, dx, ny, dy, nz, dz)
+    field = field1D.reshape((nx, ny, nz), order="F")
+    return field
+
+
+def simulate_field_using_gstools(start_seed):
+    # pylint: disable=no-member,
```

Comment:
Disabling `pylint` here is not necessary I think.

Reply:
Fixed